### PR TITLE
Add ability to change Ingress/Route monitor type

### LINKED ISSUE #699: ingress health monitor incorrect when using re-encrypt
#### Description
Using Kubernetes Ingress with re-encrypt does not use the proper type of monitor.

A "http" monitor is created instead of a "https" monitor.  Ideally a new monitor should be created that uses the same serverssl profile that was specified in the annotation.

#### Kubernetes Version
1.7.12
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.11", GitCommit:"b13f2fd682d56eab7a6a2b5a1cab1a3d2c8bdd55", GitTreeState:"clean", BuildDate:"2017-11-25T18:34:52Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.12", GitCommit:"3bda299a6414b4866f179921610d6738206a18fe", GitTreeState:"clean", BuildDate:"2017-12-29T08:39:49Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

#### Controller Version
1.5.1

#### BIG-IP Version
13.1.0.7

#### Diagnostic Information
Ingress:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: blue-green-ingress-tls
  annotations:
    virtual-server.f5.com/ip: "10.1.10.82"
    virtual-server.f5.com/https-port: "443"
    virtual-server.f5.com/ssl-redirect: "false"
    ingress.kubernetes.io/allow-http: "false"
    virtual-server.f5.com/clientssl: "/Common/_wildcard_.f5demo.com_clientssl"
    virtual-server.f5.com/serverssl: "/Common/serverssl"
    virtual-server.f5.com/secure-serverssl: "false"
    virtual-server.f5.com/partition: "kubernetes"
    virtual-server.f5.com/health: |
      [
        {
          "path":     "blue.f5demo.com/",
          "send":     "GET / HTTP/1.0",
          "interval": 5,
          "timeout":  15
        }, {
          "path":     "green.f5demo.com/",
          "send":     "GET / HTTP/1.0",
          "interval": 5,
          "timeout":  15
        }
      ]    
    kubernetes.io/ingress.class: "f5"
spec:
  tls:
  - secretName: /Common/_wildcard_.f5demo.com_clientssl
 
  rules:
  - host: blue.f5demo.com
    http:
      paths:
      - backend:
          serviceName: node-blue
          servicePort: 443
  - host: green.f5demo.com
    http:
      paths:
      - backend:
          serviceName: node-green
          servicePort: 443
```
Monitor created (note the type):
```
ltm monitor http ingress_default_node-blue_0_http {
    adaptive disabled
    defaults-from /Common/http
    destination *:*
    interval 5
    ip-dscp 0
    partition kubernetes
    recv none
    recv-disable none
    send "GET / HTTP/1.0"
    time-until-up 0
    timeout 15
}
````
Monitor log (note the 400 error):
```
[1][9781] 2018-05-25 02:28:28.181640: ID 82    :(_recv_active_service_ping): rcvd 557 bytes: -->HTTP/1.1 400 Bad Request\x0d\x0aDate: Fri, 25 May 2018 02:28:28 GMT\x0d\x0aServer: Apache/2.4.29 (Unix) LibreSSL/2.5.5\x0d\x0aContent-Length: 362\x0d\x0aConnection: close\x0d\x0aContent-Type: text/html; charset=iso-8859-1\x0d\x0a\x0d\x0a<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\x0a<html><head>\x0a<title>400 Bad Request</title>\x0a</head><body>\x0a<h1>Bad Request</h1>\x0a<p>Your browser sent a request that this server could not understand.<br />\x0aReason: You're speaking plain HTTP to an SSL-enabled server port.<br />\x0a Instead use the HTTPS scheme to access this URL, please.<br />\x0a</p>\x0a</body></html>\x0a<-- [ tmm?=false td=true tr=false addr=::ffff:10.244.1.252:443 srcaddr=::ffff:10.244.30.15%0:49028 ]
```

### PR DESCRIPTION
**Pull request by sjberman**
Health monitor type for Ingress/Route annotations was hard coded to http. Http will remain the default, however a user can now specify the type of monitor to use in the annotation.

Closes #699

### CODE REVIEW

File: pkg/appmanager/resourceConfig.go [hunk 1]
```diff
@@ -2209,8 +2211,7 @@ func (rc *ResourceConfig) RemoveMonitor(pool, monitor string) bool {
 		}
 	}
 	for i, mon := range rc.Monitors {
-		name := strings.Split(monitor, "/")[2]
-		if mon.Name == name {
+		if strings.Contains(monitor, mon.Name) {
```
Comment (russokj) [inline pkg/appmanager/resourceConfig.go]: Why was this needed for the change?

Comment (sjberman) [reply pkg/appmanager/resourceConfig.go]: If a monitor doesn't exist for a pool, then the "monitor" string won't get set. The array index would crash. I could make this safer by wrapping this for loop in a conditional; whether monitor is empty or not.
